doc/api/events.md

```diff
@@ -155,9 +155,63 @@ myEmitter.emit('error', new Error('whoops!'));
 // Prints: whoops! there was an error
 ```
 
+## AutoCatch of Promise rejection
+
+Using async functions in event handlers is problematic, because it would
+lead to `'unhandledRejection'` in case of an exception
+
+```js
+const ee = new EventEmitter();
+ee.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+```
+
+The `autoCatch` option in the `EventEmitter` constructor or the global
+setting change this behavior, installing a `.catch()` handler on the
+`Promise`. This `catch` handler routes the exception to the
+[`Symbol.for('nodejs.rejection')`][rejection] event if there is a listener, or to
+[`'error'`][error] if there is none.
+
+```js
+const ee1 = new EventEmitter({ autoCatch: true });
+ee1.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+
+ee1.on('error', console.log);
+
+const ee2 = new EventEmitter({ autoCatch: true });
+ee2.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+
+ee2.on(Symbol.for('nodejs.rejection'), console.log);
```

Comment:
I think that if we keep this symbol it's likely better to keep it on events rather than the global symbol registry like symbol.promisify.custom unless there is a specific need for it to behave well across realms.

Reply:
Keeping it on the global symbol registry allows it to be used browser side as well, which is going to be important for things like `readable-stream` and the various browser ports of EventEmitter. That said, exposing the symbol as a convenience property makes sense... e.g. 

```js
const { EventEmitter, rejection } = require('events')

ee2.on(rejection, console.log)

// would be equivalent to

ee2.on(Symbol.for('nodejs.rejection'), console.log)

// either would work
```

Reply:
I'm a bit unsure about exporting it as a symbol, mainly because we have a global switch at `events.captureRejections`, and the `Symbol` at `events.rejection`. I think this can get confusing very quickly.